metadata-ws/src/main/java/uk/ac/ebi/ampt2d/metadata/aop/StudyReleaseDateAspect.java

```diff
@@ -42,39 +41,44 @@
      * @return the return object from join point method execution
      * @throws Throwable
      */
-    @Around("execution(* uk.ac.ebi.ampt2d.metadata.persistence.repositories.StudyRepository.findAll(..))")
-    public Object filterOnReleaseDateAdviceFindAll(ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
-        Object[] args = proceedingJoinPoint.getArgs();
-        Predicate predicate = (Predicate) args[0];
-
-        QStudy study = QStudy.study;
-        Predicate dateRestrictedPredicate = study.releaseDate.between(null, LocalDate.now()).and(predicate);
 
-        args[0] = dateRestrictedPredicate;
-
-        return proceedingJoinPoint.proceed(args);
+    @Around("execution(* uk.ac.ebi.ampt2d.metadata.persistence.repositories.*.findAll(..))")
+    public Object filterOnReleaseDateAdviceFindAll(ProceedingJoinPoint proceedingJoinPoint) throws Throwable {
+        Iterable<Object> results = (Iterable<Object>) proceedingJoinPoint.proceed();
+        Iterator<Object> i = results.iterator();
```

Comment:
Yes, because the `findAll` methods of all repositories return:
* Either **`Iterable`**,
* Or `Page`, which extends `Slice`, which extends **`Iterable`**.

So we're always safe to use the `Iterable` methods.